src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1010,24 +1010,28 @@ private void WindowClosed(object sender, EventArgs e)
         // passes it to thecurrent workspace
         void DynamoView_KeyDown(object sender, KeyEventArgs e)
         {
-            if (e.Key == Key.Escape)
-                dynamoViewModel.BackgroundPreviewViewModel.CanNavigateBackground = true;
+            if (e.Key == Key.Escape && this.IsMouseOver)
+            {
+                dynamoViewModel.BackgroundPreviewViewModel.NavigationKeyIsDown = true;
+                e.Handled = true;
+            }
         }
 
         void DynamoView_KeyUp(object sender, KeyEventArgs e)
         {
             if (e.Key == Key.Escape && dynamoViewModel.BackgroundPreviewViewModel.CanNavigateBackground)
             {
-                dynamoViewModel.BackgroundPreviewViewModel.CanNavigateBackground = false;
+                dynamoViewModel.BackgroundPreviewViewModel.NavigationKeyIsDown = false;
                 dynamoViewModel.EscapeCommand.Execute(null);
+                e.Handled = true;
```

Comment:
Will this impact the processing of the ESC key message? For example, if you double click in the canvas, it will add a code block node, and if you press ESC, the new node will be deleted and the focus will go to the search box.